core/server/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.InvalidPathException;
+
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class represents a temporary {@link LockedInodePath}. This {@link LockedInodePath} will
+ * not unlock the inodes on close. This {@link LockedInodePath} can set any descendant (does not
+ * have to be immediate).
+ *
+ * This is useful for being able to pass in descendant inodes to methods which require a
+ * {@link LockedInodePath}, without having to re-traverse the inode tree, and re-acquire locks.
+ * This allows methods to operate on descendants associated with an existing
+ * {@link LockedInodePath}.
+ */
+@ThreadSafe
```

Comment:
would an iterator over the children of the LockedInodePath leaf address all use cases of this class? if so, please add a TODO to consider replacing this class with an iterator for LockedInodePath